packages/eslint-plugin/tests/rules/no-use-before-define.test.ts

```diff
@@ -219,7 +219,29 @@ type Foo = string | number;
       `,
       options: [{ typedefs: false }],
     },
+    // https://github.com/typescript-eslint/typescript-eslint/issues/2572
+    {
+      code: `
+interface Bar {
+  type: typeof Foo.FOO;
+}
+
+class Foo {
+  public static readonly FOO = '';
+}
+      `,
+      options: [{ ignoreTypeReferences: true }],
+    },
+    {
+      code: `
+const foo = 2;
 
+interface Bar {
+  type: typeof foo;
+}
```

Comment:
This test is passing because `foo` is in fact defined before it's used, not because of your changes.

If you move `foo` beneath the `interface`, I believe this test will fail right now.

Reply:
Dang it 😅
I didn't think of looping the first time because this test was passing and thought maybe the `TSTypeQuery` was always at the same depth, but the test was passing for this reason.

I've made the necessary changes.